### ISSUE TITLE
Build failure of src/test

### ISSUE THREAD

**Issue by AntonyMei** [quantum-compiler/quartz#14]
I ran into this error when trying to build test files in `src/test` in latest version of master branch:

    /usr/bin/ld: CMakeFiles/test_toffoli_flip.dir/test_toffoli_flip.o: in function `main':
    test_toffoli_flip.cpp:(.text+0x1345): undefined reference to `quartz::Graph::ccz_flip_t(quartz::Context*)'
    /usr/bin/ld: test_toffoli_flip.cpp:(.text+0x13ba): undefined reference to `quartz::Graph::to_qasm(std::__cxx11::basic_string<char, 
    std::char_traits<char>, std::allocator<char> > const&, bool, bool) const'
    collect2: error: ld returned 1 exit status
    make[2]: *** [CMakeFiles/test_toffoli_flip.dir/build.make:103: test_toffoli_flip] Error 1
    make[1]: *** [CMakeFiles/Makefile2:137: CMakeFiles/test_toffoli_flip.dir/all] Error 2
    make: *** [Makefile:103: all] Error 2

It seems that quartz's c code is modified but the test files are not updated, because previous versions of master does not have this problem.

**Comment by AntonyMei** [comment]
I was compiling this using `cmake=3.19.4` and `gcc=9.3.0`. The following steps can reproduce this issue

    cd src/test/
    mkdir build
    cd build
    cmake ..
    make

**Comment by Co1lin** [comment]
Maybe we can add a CI to ensure that the code on the master branch can be built successfully.

**Comment by jiazhihao** [comment]
Yep, we can use GitHub CI or CircleCI, whichever is easier to integrate.

**Comment by zikun-li** [comment]
I tried to reproduce the error, but on my platform the error occurs in a different file and iscaused by different `undefined reference`. So I guess it's something related to the relative path `../..` used in our `CMakeList.txt` in `src/test`. Can you try using the following script to compile?
In your `quartz` directory:
```
mkdir build
cd build
cmake ..
make
```

**Comment by AntonyMei** [comment]
I agree. Everything works well when compiling with the `CMakeList.txt` in the outermost directory. Maybe we should remove the `CMakeList.txt` in `src/test`?
